lib/parser.js

```diff
@@ -253,4 +254,28 @@ async function customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialF
     }));
   });
   await Promise.all(promisesArray);
-}
\ No newline at end of file
+}
+
+/**
+ * Triggers additional operations on the AsyncAPI components/messages like traits application, validation and conversion
+ *
+ * @private
+ * 
+ * @param  {Object} parsedJSON parsed AsyncAPI document
+ * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
+ * @param  {String} initialFormat information of the document was originally JSON or YAML
+ * @param  {Object} options Configuration options.
+ */
+async function customMessagesOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options) {
+  const promisesArray = [];
+
+  Object.entries(parsedJSON.components.messages).forEach(([messageName, message]) => {
```

Comment:
Before accesing to `parsedJSON.components.messages` please first check that `components.messages` path exists, because either `components` and `messages` in `components` are optional, something like:

```js
async function customMessagesOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options) {
  if (!parsedJSON.components || !parsedJSON.components.messages) return;
  ...your logic
}
```